lib/tasks/edition.rake

```diff
@@ -0,0 +1,12 @@
+namespace :edition do
+  desc 'Check if any deleted edition has undeleted attachments, and delete them'
+  task delete_edition_files: :environment do
+    Edition.where(state: :deleted).find_each do |edition|
+      edition.attachments.each do |attachment|
```

Comment:
You could change the query here to be `edition.attachments.where(deleted: false)` and avoid the `next if`.